### PR TITLE
[SPARK-40851][INFRA ][SQL][TESTS] Make GA run successfully with the latest Java 8/11/17

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
The main change of this pr as follows:

- Replace `Antarctica/Vostok` to `Asia/Urumqi` in Spark code
- Replace `Europe/Amsterdam` to `Europe/Brussels` in Spark code
- Regenerate `gregorian-julian-rebase-micros.json` using generate 'gregorian-julian-rebase-micros.json' in `RebaseDateTimeSuite` with Java 8u352
- Regenerate `julian-gregorian-rebase-micros.json` using generate 'julian-gregorian-rebase-micros.json' in RebaseDateTimeSuite with Java 8u352
- Revert change of SPARK-40846

### Why are the changes needed?
Make GA run successfully with the latest Java 8/11/17:

- Java 8u352
- Java 11.0.17
- Java 17.0.5

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

- Pass GitHub Actions
- Manual test: the following commands can test pass with Java 8u345, 8u352, 11.0.16, 11.0.17, 17.0.4 and 17.0.5
   - `build/sbt "catalyst/test"`
   - `build/sbt "sql/testOnly *SQLQueryTestSuite -- -t \"timestamp-ltz.sql\""`
   - `build/sbt "sql/testOnly *SQLQueryTestSuite -- -t \"timestamp-ntz.sql\""`

### CODE REVIEW

**Comment by LuciferYang** [comment]
The replacement of Zone refers to the `aliases.txt` of Java's own test:

- `Antarctica/Vostok -> Asia/Urumqi`:  https://github.com/openjdk/jdk8u-dev/blob/46da2c31efef73b7c2914c8d252554556145b00b/jdk/test/java/util/TimeZone/TimeZoneData/aliases.txt#L40
- `Europe/Amsterdam -> Europe/Brussels`: https://github.com/openjdk/jdk8u-dev/blob/46da2c31efef73b7c2914c8d252554556145b00b/jdk/test/java/util/TimeZone/TimeZoneData/aliases.txt#L68

File: sql/catalyst/src/main/resources/julian-gregorian-rebase-micros.json [hunk 1]
```diff
@@ -5,14 +5,14 @@
 }, {
   "tz" : "Africa/Accra",
   "switches" : [ -62135769600, -59006448000, -55850688000, -52694928000, -46383408000, -43227648000, -40071888000, -33760368000, -30604608000, -27448848000, -21137328000, -17981568000, -14825808000, -12219292800, -2208988800 ],
-  "diffs" : [ 172852, 86452, 52, -86348, -172748, -259148, -345548, -431948, -518348, -604748, -691148, -777548, -863948, 52, 0 ]
+  "diffs" : [ 173768, 87368, 968, -85432, -171832, -258232, -344632, -431032, -517432, -603832, -690232, -776632, -863032, 968, 0 ]
```
Comment (LuciferYang) [inline sql/catalyst/src/main/resources/julian-gregorian-rebase-micros.json:8]: Regenerate `julian-gregorian-rebase-micros.json` using `generate 'julian-gregorian-rebase-micros.json'` in `RebaseDateTimeSuite` with Java 8u352

**Comment by HyukjinKwon** [comment]
cc @MaxGekk

**Review by cloud-fan — APPROVED** [review]

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
All GAs passed at least once:
Screenshot 2022-10-21 at 09 55 20

Merging to master. Thank you, @LuciferYang and @cloud-fan @HyukjinKwon for review.

**Comment by LuciferYang** [comment]
Thanks @MaxGekk @cloud-fan @HyukjinKwon and @wangyum

**Comment by srowen** [comment]
@LuciferYang and everyone, we need to port this back to 3.3 and 3.2 right? b/c it will affect those branches, which do not pin Java 8. For example I think this backport is failing with the same issue: https://github.com/bjornjorgensen/spark/actions/runs/3304235420/jobs/5455772005

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala [hunk 2]
```diff
@@ -937,16 +937,16 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers with SQLHelper {
       ("1970-01-01T00:00:00", "UTC") -> ("1969-12-31T16:00:00", "America/Los_Angeles"),
       ("2021-12-05T22:00:00", "Europe/Moscow") -> ("2021-12-06T00:00:00", "Asia/Yekaterinburg"),
       ("2021-12-06T00:01:02.123456", "Asia/Yekaterinburg") ->
-        ("2021-12-05T20:01:02.123456", "Europe/Amsterdam"),
+        ("2021-12-05T20:01:02.123456", "Europe/Brussels"),
```
Comment (HyukjinKwon) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala:940]: @LuciferYang mind creating a cherry-pick to other branches too?

Comment (LuciferYang) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala:940]: Will do this today @srowen @HyukjinKwon

Comment (LuciferYang) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala:940]: - branch-3.3: https://github.com/apache/spark/pull/38363
- branch-3.2: https://github.com/apache/spark/pull/38365

**Pull request closed without merging**